### DISCUSSION TITLE
React integration without mobx-react

### DISCUSSION THREAD

**Discussion by shiningjason** [mobxjs/mobx#2863] [Q&A] [answered]
Hi, I'm new to mobx and use it with React.

Can I use the following hook (`useSelector`) to subscribe observable?  
Or is there any side effect?

```js
function useSelector(selector) {
  const [state, setState] = useState(selector());
  useEffect(() => autorun(() => setState(selector())), []);
  return state;
}

function App() {
  const count = useSelector(() => store.count);
  return count;
}

class CounterStore {
  @observable
  count = 0;

  constructor() {
    makeObservable(this);
  }
}
```

[CodeSandbox: sample code](https://codesandbox.io/s/use-mobx-selector-1w17y?file=/src/index.js)

**Comment by danielkcz** [comment, marked as answer]
Have a look at mobx-react-lite source code for inspiration. It's fairly really simple with hooks. But that example of yours is generally a discouraging pattern. Subscribe to observables within components if you want to save yourself problems later. There are other similar topics/issues discussing it.

**Reply by shiningjason** [reply]
Thanks for your answer!
